block/manager.go

```diff
@@ -101,11 +101,23 @@ type Manager struct {
 
 // getInitialState tries to load lastState from Store, and if it's not available it reads GenesisDoc.
 func getInitialState(store store.Store, genesis *cmtypes.GenesisDoc) (types.State, error) {
+	b, err := store.GetBlock(uint64(genesis.InitialHeight))
+	if err != nil {
+		// we've never seen a block at InitialHeight.
+		// assume we're syncing from a trusted snapshot
+		return types.NewFromGenesisDoc(genesis)
+	}
+	if !bytes.Equal(b.SignedHeader.AppHash, genesis.AppHash.Bytes()) {
```

Comment:
Can these actually ever be different? `getInitialState` is called at the start of `NewManager`. If a node normally restarts, do we hit these points?

Reply:
if the user stops the node and restarts it with a different genesis, overriding an accepted block